creusot/src/translation/function.rs

```diff
@@ -339,30 +534,79 @@ impl<'body, 'tcx> BodyTranslator<'body, 'tcx> {
             return;
         }
 
-        let resolved_between = pred_blocks
+        let mut resolved_between = pred_blocks
             .iter()
-            .map(|pred| resolver.resolved_locals_between_blocks(*pred, bb))
+            .map(|&pred| resolver.resolved_places_between_blocks(pred, bb))
             .collect::<Vec<_>>();
 
+        for (pred, resolved) in iter::zip(pred_blocks, resolved_between.iter_mut()) {
+            // We do not need to resolve move path that we know are inactive
+            // because of a preceding switch.
+
+            let bbd = &self.body.basic_blocks[*pred];
+            let _: Option<()> = try {
+                let discr_pl = discriminator_for_switch(bbd)?;
+                let discr_mp = if let LookupResult::Exact(mp) =
+                    self.move_data().rev_lookup.find(discr_pl.as_ref())
+                {
+                    mp
+                } else {
+                    continue;
+                };
+                let adt = discr_pl.ty(self.body, self.tcx).ty.ty_adt_def()?;
+                let targets =
+                    if let TerminatorKind::SwitchInt { targets, .. } = &bbd.terminator().kind {
+                        targets
+                    } else {
+                        unreachable!()
+                    };
+                if targets.otherwise() == bb {
+                    continue;
+                }
+
+                let mut inactive_mps = self.empty_bitset();
+                on_all_children_bits(self.move_data(), discr_mp, |mpi| {
+                    inactive_mps.insert(mpi);
+                });
```

Comment:
Why is the discriminator `MovePlace` inactive? I don't really understand this bit of logic.

Reply:
Well, it's not necessarilly. We initialized all the children of the discriminiator to true, and the remove whatever is active.